index-migration/indexMigration.py

```diff
@@ -0,0 +1,66 @@
+import subprocess
+import time
+
+def check_directory(filename):
+    """
+    To verify if filename is in current directory
+    """
+    ls_out=subprocess.Popen(['ls'], stdout=subprocess.PIPE)
+    directories = ls_out.stdout.read()
+    return filename in directories
+
+def backup_db():
+    """
+    To backup mongo databases and save it to a directory called dump.bak
+    """
+    if check_directory("dump"):
+       subprocess.Popen(['rm -rf dump*'], shell=True)
+       time.sleep(1)
+       assert not check_directory("dump"), "The dump file is not deleted successfully"
+
+    subprocess.Popen(['mongodump'], stdout=subprocess.PIPE)
+    time.sleep(3)
```

Comment:
I think using `subprocess.call` might make this sleep unnecessary.  We need to be really certain that the backup is successful before proceeding with the migration to avoid losing production data.